addons/web/controllers/main.py

```diff
@@ -325,18 +325,29 @@ def set_cookie_and_redirect(redirect_url):
     return redirect
 
 def clean_action(action, env):
-    action.setdefault('flags', {})
     action_type = action.setdefault('type', 'ir.actions.act_window_close')
     if action_type == 'ir.actions.act_window':
         action = fix_view_modes(action)
 
     # When returning an action, only few information are really usefull
-    return {
+    readable_fields = env[action['type']]._get_readable_fields()
+    action_type_fields = env[action['type']]._fields.keys()
+
+    cleaned_action = {
         field: value
         for field, value in action.items()
-        if field in env[action['type']]._get_readable_fields()
+        # keep allowed fields and custom properties fields
+        if field in readable_fields or field not in action_type_fields
     }
 
+    # Warn the presence of custom properties fields in the action, as their use should be discouraged.
```

Comment:
nit: grammar + line length
```suggestion
    # Warn about custom properties fields, because use is discouraged
```